lib/ssh_scan/scan_engine.rb

```diff
@@ -9,11 +9,34 @@ def scan_target(socket, opts)
       target, port = socket.chomp.split(':')
       policy = opts[:policy_file]
       timeout = opts[:timeout]
+      result = []
 
-      client = SSHScan::Client.new(target, port, timeout)
-      client.connect()
-      result = client.get_kex_result()
-      return result if result.include?(:error)
+      if target.fqdn?
+        if target.resolve_fqdn_as_ipv6.nil?
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv4.to_s, port, timeout)
+          client.connect()
+          result = client.get_kex_result()
+          result[:hostname] = target
+          return result if result.include?(:error)
+        else
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv6.to_s, port, timeout)
```

Comment:
I think we might need some localized code to determine if the scanner itself can route v6 traffic to the endpoint and maybe a more explicit error that says "Hey, I can only scan v4 hosts" or something.